llzk_backend/src/module.rs

```diff
@@ -61,6 +59,10 @@ struct DeclarationInfo<'ctx> {
 impl<'ctx> DeclarationInfo<'ctx> {
     /// Visit a statement and populate this `DeclarationInfo` with any declarations found.
     ///
+    /// TODO: This currently visits only top-level statements within the template body. However,
+    /// since circom 2.1.5, signal declarations are allowed inside of blocks and known-condition if
+    /// statements. Those nested declarations are not currently processed here.
+    ///
```

Comment:
Will not be addressed in this PR. Need to create issue.

Reply:
I'd note on that issue that that feature should be on only if `>=2.1.5`, which may require some sort of dynamic selection of features based on the target version, which may be worth its own issue